test/unit/producer.test.ts

```diff
@@ -0,0 +1,58 @@
+import { expect } from "chai"
+import { randomUUID } from "crypto"
+import { connect } from "../../src"
+import { Rabbit } from "../support/rabbit"
+
+describe("Producer", () => {
+  const rabbit = new Rabbit()
+  const testStreamName = "test-stream"
+  const publisherRef = randomUUID()
```

Comment:
This publishRef remains the same for every test, I suggest to put it or in beforeEach or in every test